<?xml version="1.0" encoding="UTF-8" standalone="no" ?>
<launch>
	<arg name="model_file" 		default="/srv/data/gait1992/residual_reduction_algorithm/model_adjusted.osim" />
	<arg name="get_second_label" 	default="true"/>
	<arg name="left_foot_tf_name" 	default="left_foot_forceplate" />
	<arg name="right_foot_tf_name" 	default="right_foot_forceplate" />
	<arg name="use_exact_sync"	default="false"/>

        <node pkg="osrt_ros" type="id" 	name="id_node" output="screen">
                <param name="model_file" 		value="$(arg model_file)" />
                <param name="get_second_label" 		value="$(arg get_second_label)" />
		<param name="left_foot_tf_name"		value="$(arg left_foot_tf_name)" />
		<param name="right_foot_tf_name"	value="$(arg right_foot_tf_name)" />
		<param name="grf_reference_frame"        value="subject_opensim"/>
		<param name="use_exact_sync"		value="$(arg use_exact_sync)"/>
		<remap from="/id_node/input_filtered" to="/ik/output_filtered"/>
		<remap from="/in_labels" to="/ik/outlabels"/>
		<!--remap from="/id_node/input2" 	to="/inverse_kinematics_from_file/r_data2"/>
		<remap from="/in_labels2" 	to="/inverse_kinematics_from_file/out_labels"/ -->
		<remap from="/id_node/input2" 	to="/grf_node/output"/>
		<remap from="/in_labels2" 	to="/grf_node/out_labels"/>
		
		<remap from="/id_node/left_wrench" 	to="/grf_left/wrench"/>
		<remap from="/id_node/right_wrench" 	to="/grf_right/wrench"/>
		<param name="grf_right_point_identifier" 		value="ground_force_p" />
                <param name="grf_right_force_identifier" 		value="ground_force_v" />
                <param name="grf_right_torque_identifier" 		value="ground_torque_" />
                <param name="grf_right_apply_to_body" 			value="calcn_r" />
                <param name="grf_right_force_expressed_in_body" 	value="ground" />
                <param name="grf_right_point_expressed_in_body" 	value="ground" />
		
		<param name="grf_left_point_identifier" 		value="1_ground_force_p" />
                <param name="grf_left_force_identifier" 		value="1_ground_force_v" />
                <param name="grf_left_torque_identifier" 		value="1_ground_torque_" />
                <param name="grf_left_apply_to_body" 			value="calcn_l" />
                <param name="grf_left_force_expressed_in_body" 		value="ground" />
		<param name="grf_left_point_expressed_in_body" 		value="ground" />
		
        </node>

</launch>
